CarRR-kjar/src/main/resources/recommendation_rules/rental_history_recommendation_rules.drl

```diff
@@ -142,3 +159,91 @@ lock-on-active true
     then
         modify ($recommendations) {addRentalHits($vehicle, $rentalFuel.get($fuel))}
 end
+
+
+rule "Rental owner set status rule"
+agenda-group "rentals"
+salience 10
+no-loop true
+lock-on-active true
+    when
+        Rental($owner: owner)
+        $user: User(noOfRentals >= 10) from $owner
+    then
+        System.out.println("------------- RENTAL RULES: Assing owner status. -------------");
+
+        modify ($user) {setStatus(OwnerStatus.GOLD))}
+end
+
+rule "Rental - Save recommendations and remove facts from session"
+agenda-group "rentals"
+salience 2
+    when
+        not Bundle()
+    	$c: Customer($recommendations: recommendations)
+        $r: Recommendations() from $recommendations
+    then
+        System.out.println("------------- RENTAL RULES: Removing customer and recommendations -------------");
+    	System.out.println($r);
+        recommendationsRepo.save($r);
+        customerRepository.save($c);
+    	delete($c);
+    	delete($r);
+end
+
+rule "Rental - Remove vehicle from session"
+agenda-group "rentals"
+salience 1
+    when
+        not Bundle()
+    	$v: Vehicle()
+    then
+        System.out.println("------------- RENTAL RULES: Removing vehicle -------------");
+    	delete($v);
+end
+
+rule "Rental - increase number of finished recommendation rules"
+agenda-group "rentals"
+no-loop true
+salience 1
+    when
+        $b: Bundle($finishedRules: finishedRules, $vehicleID: vehicleID)
+        $v: Vehicle($id: id)
+    then
+        if($vehicleID == $id){
+            System.out.println("------------- RENTAL RULES: Search recommendation rules done - increase count -------------");
+            modify ($b) {increaseFinishedRules()};
+        }
+end
+
+rule "Rental - remove vehicle from session if all done"
+agenda-group "rentals"
+salience 0
+    when
+        $b: Bundle($finishedRules: finishedRules, $vehicleID: vehicleID)
+     	$v: Vehicle($id: id)
+    then
+        System.out.println($finishedRules);
+        if($finishedRules >= 3  && $vehicleID == $id){
+             System.out.println("------------- RENTAL RULES: Removing vehicles if all recommendation rules are done-------------");
+             delete($v);
+        }
+end
+
+rule "Rental - save recommendations and remove facts from session if all done"
+agenda-group "rentals"
+salience 0
+    when
+        $b: Bundle($finishedRules: finishedRules, $vehicleID: vehicleID)
+    	$c: Customer($recommendations: recommendations)
+        $r: Recommendations() from $recommendations
+        $v: Vehicle($id: id)
+    then
+        System.out.println($finishedRules);
+       if($finishedRules >= 3 && $vehicleID == $id){
```

Comment:
Maybe we can put this in WHEN part of the rule, don't know why we didn't think of this yesterday...